packages/uma-auth-client/src/utils/getNwcUmaDomain.ts

```diff
@@ -0,0 +1,8 @@
+export const getNwcUmaDomain = (uma: string) => {
+  const umaDomain = uma.split("@")[1];
+  const isLocal =
+    umaDomain.startsWith("localhost:") || umaDomain.includes(".local:");
+  const protocol = isLocal ? "http" : "https";
+  const nwcPrefix = !isLocal ? "nwc." : "";
+  return `${protocol}://${nwcPrefix}${umaDomain}`;
```

Comment:
This isn't right - all nwc domain routes should come from the uma-configuration as needed. I don't think we should ever assume static subdomains or routes if we can avoid it.

Reply:
hm should we add the base url then or just get the base from the /oauth/auth or token urls?

Reply:
getting from auth url for now

Reply:
I'm confused what you need the base for in general. I think we should be flexible enough such that different endpoints can be hosted at different domains even, if we wanted them to be.